libavfilter/signature_lookup.c

```diff
@@ -244,9 +244,9 @@ static MatchingInfo* get_matching_parameters(AVFilterContext *ctx, SignatureCont
                     if (pairs[i].b[j] != pairs[k].b[l]) {
                         /* linear regression */
                         m = (pairs[k].b_pos[l]-pairs[i].b_pos[j]) / (k-i); /* good value between 0.0 - 2.0 */
-                        framerate = (int) m*30 + 0.5; /* round up to 0 - 60 */
+                        framerate = nearbyint(m*30 + 0.5); /* round up to 0 - 60 */
```

Comment:
Really nice fix!

Going by the paper - the hough transform is used to calculate the framerate and time-shift changes between the two video samples by voting across all fine signature pairs that are within a threshold `th_xh` distance of each other. 

I think the original author was casting a float to an int after adding 0.5 to round it to nearest integer. But because they missed the brackets around their float expression `(int) (m*30 + 0.5)` what ended up happening was `m` got converted to `int` before rest of the additions. As the end result the code before this change was erroneously only considering framerates 0, 30 and 60 - which was unintended.

After your fix using `nearbyint` - it'll now vote for all integer values between 0-60 for framerate, so that should really improve accuracy I think!

One small suggestion - we should remove the `0.5` now as we're using `nearbyint` which by default rounds to nearest integer. If we add `0.5` we'll actually do `ceil` operation, which is better than the bug before, but still not completely accurate here.
```suggestion
                        framerate = nearbyint(m*30); /* round up to 0 - 60 */
```

Reply:
Also I think it makes sense to move this into a separate commit, so that we can send this patch upstream to FFmpeg :)